tests/unit/fem/ptest_tdofs.cpp

```diff
@@ -0,0 +1,1012 @@
+   {
+      MPRINT_MASTER(rank,"Finding boundary element tdofs");
+
+      const int dim = m_mesh->Dimension();
+      const int order = 0;
+      const int NBE = m_mesh->GetNBE();
+
+      int NCodimFaces = 0;
+      switch (dim)
+      {
+         case 1: NCodimFaces = m_mesh->GetNV(); break;
+         case 2: NCodimFaces = m_mesh->GetNEdges(); break;
+         case 3: NCodimFaces = m_mesh->GetNFaces(); break;
+         default: FAIL("Dim must be 1,2 or 3."); break;
+      }
+
+      auto* fec = new mfem::RT_FECollection(order, dim);
+      mfem::ParFiniteElementSpace fes(m_mesh, fec);
+      mfem::Array<int> be_tdof(NBE);
+      be_tdof = UNITIALIZED_VAL;
+
+      mfem::Array<int> idofs;
+
+      std::stringstream sstr;
+      for (int be=0; be< NBE; ++be)
+      {
+         // Find tdof of mesh element corresponding to the boundary element
+         auto elem_id = m_mesh->GetBdrElementEdgeIndex(be);
+         CHECK( elem_id >= 0);
+         CHECK( elem_id < NCodimFaces);
+
+         switch (dim)
+         {
+            case 1: fes.GetVertexDofs(elem_id, idofs); break;
+            case 2: fes.GetEdgeDofs(elem_id, idofs); break;
+            case 3: fes.GetFaceDofs(elem_id, idofs); break;
+         }
+         CHECK( idofs.Size() == 1);
+
+         auto ldof = decodeDof(idofs[0]);
+         CHECK( ldof >= 0);
+
+         // Check local tdof?
+         // auto ltdof = fes.GetLocalTDofNumber(ldof);
+
+         auto tdof = fes.GetGlobalTDofNumber(ldof);
+         CHECK( tdof >= 0);
+
+         be_tdof[be] =tdof;
+
+         // generate some debug output
+         if (m_verbose)
+         {
+            sstr<< std::endl <<"\t"
+                <<"{be:" << be
+                <<",l:" << ldof
+                <<",T_l:" << fes.GetLocalTDofNumber(ldof)
+                <<",T_g:" << tdof
+                <<"}";
+         }
+      }
+      MPRINT(rank, sstr.str());
+
+      // check that all elements have valid tdofs
+      checkDOFs(be_tdof, "boundary element");
+
+      delete fec;
+   }
+
+   mfem::ParMesh* m_mesh;
+   static constexpr int UNITIALIZED_VAL = -100;
+   int rank;
+   bool m_verbose;
+};
+
+} // end anonymous namespace
+
+
+TEST_CASE("Segment mesh tdofs",
+          "[ParFESpace]"
+          "[Hypre]"
+          "[TDofs]")
+{
+   int rank;
+   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+   MPRINT_MASTER(rank, "=== Checking TDOFs for Segment mesh ===");
+
+   const int attrib = 1;
+   mfem::Mesh* mesh = nullptr;
+   mfem::ParMesh* pmesh = nullptr;
+
+   // Build a simple single element segment mesh
+   const int dim = 1, nv = 2, ne = 1, nb = 0, sdim = 2;
+   {
+      mesh = new mfem::Mesh(dim, nv, ne, nb, sdim);
+
+      mesh->AddVertex(mfem::Vertex(0.,0.)());
+      mesh->AddVertex(mfem::Vertex(1.,0.)());
+
+      int idx[2] = {0,1};
+      mesh->AddElement(new mfem::Segment(idx, attrib));
+
+      mesh->FinalizeMesh();
+
+      CHECK( mesh->GetNV() == nv);
+      CHECK( mesh->GetNE() == ne);
+      //CHECK( mesh->GetNBE() == 2);
```

Comment:
Do segment meshes have boundary elements?  

This call (``mesh->GetNBE()``) was returning zero.